### PR TITLE
APEXCORE-475 1st part, as discussed in the jira. This will make the c…

### PR DESCRIPTION
**Pull request by sandeshh**
…lient the forward compatible, then in the next release we can change the type to of the applicationt to ApacheApex

### CODE REVIEW

**Comment by sandeshh** [comment]
@davidyan74 please review

File: engine/src/main/java/com/datatorrent/stram/StramClient.java [hunk 1]
```diff
@@ -95,7 +95,9 @@
 public class StramClient
 {
   private static final Logger LOG = LoggerFactory.getLogger(StramClient.class);
-  public static final String YARN_APPLICATION_TYPE = "DataTorrent";
+  public static final String YARN_APPLICATION_TYPE_DEPRECATED = "DataTorrent";
+  public static final String YARN_APPLICATION_TYPE = "ApacheApex";
```
Comment (chinmaykolhatkar) [inline engine/src/main/java/com/datatorrent/stram/StramClient.java:98]: Apologies if this feedback is any late, but I suggest not to attach any organization's name to yarn application type or any such string. Reason being same as why we have to change it now.
Suggesting to make it "Apex" OR "ApexStreaming"

Comment (sandeshh) [reply engine/src/main/java/com/datatorrent/stram/StramClient.java:98]: Let us stick with "ApacheApex", as
1. we need unique name for the Application type.
2. All the packages are(will) using org.apache.apex.. should we change them as well?
3. If at all the we have to change for a reason you mentioned then we have bigger things to worry than the type of an application

Comment (tweise) [reply engine/src/main/java/com/datatorrent/stram/StramClient.java:98]: Have you checked what other Apache projects are using? I'm fine with ApacheApex and I suspect it's inline with the ASF branding guidelines.

**Comment by tweise** [comment]
I think introducing lookup for 3.5 and then actually switch in 3.6 is a good plan.

File: engine/src/main/java/com/datatorrent/stram/StramClient.java (hunk 1 above)
Comment (sandeshh) [reply engine/src/main/java/com/datatorrent/stram/StramClient.java:98]: Other names that I saw are of the form "Apache xyz"

**Comment by tweise** [comment]
@sandeshh please rebase and I will merge. We should keep the JIRA open though until this is fully implemented..

**Comment by sandeshh** [comment]
@tweise rebase done. Yes, let us keep the jira open.

**Pull request merged**